src/renderers/dom/client/ReactMount.js

```diff
@@ -468,6 +469,14 @@ var ReactMount = {
     var containerHasNonRootReactChild = hasNonRootReactChild(container);
 
     if (__DEV__) {
+      var validElement = typeof nextElement.type === 'string' ?
```

Comment:
Per @jimfb new errors should be implemented within devtools (you can reference https://github.com/facebook/react/pull/7040 as a template)